packages/client/utils/AtlassianManager.ts

```diff
@@ -613,7 +613,38 @@ export default abstract class AtlassianManager {
       const projectFilter = projectKeys.length
         ? `project in (${projectKeys.map((val) => `\"${val}\"`).join(', ')})`
         : ''
-      const textFilter = queryString ? `text ~ \"${queryString}\"` : ''
+
+      var issueKeyJQL = ''
+      if (queryString) {
+        const maybeIssueKeys = queryString.split(/[\s,]+/)
+        if (maybeIssueKeys.length > 0) {
+          const validIssueKeys = maybeIssueKeys.map((maybeIssueKey) => {
+            const match = maybeIssueKey.match(/(?<projectKey>[A-Za-z][A-Za-z_0-9]+)*-*(?<issueNumber>\d+)/)
+            if (!match || !match.groups) return ""
+
+            const {projectKey: maybeProjectKey, issueNumber: maybeIssueNumber} = match.groups
+            if (maybeProjectKey && !maybeIssueNumber) return ""
+            else if (maybeProjectKey && maybeIssueNumber) {
+              if (projectKeys.length == 0 || (projectKeys.length != 0 && projectKeys.includes(maybeProjectKey.toUpperCase()))) {
+                return `${maybeProjectKey}-${maybeIssueNumber}`
+              } else {
+                return ""
+              }
+            } else if (!maybeProjectKey && maybeIssueNumber) {
+              if (projectKeys.length == 0) {
+                return ""
+              } else {
+                return projectKeys.map((projectKey) => `${projectKey}-${maybeIssueNumber}`)
+              }
+            } else {
+              return ""
+            }
+          }).flat().filter(String).map((issueKey) => `\"${issueKey.toUpperCase()}\"`)
+          issueKeyJQL = validIssueKeys.length > 0 ? ` OR issueKey in (${validIssueKeys.join(', ')})` : ''
+        }
+      }
```

Comment:
+1 this could be a separate function with a proper name, it might make the purpose of this part of the code clear